azkaban-web-server/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -0,0 +1,113 @@
+package azkaban.imagemgmt.daos;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.db.SQLTransaction;
+import azkaban.imagemgmt.models.ImageOwnership;
+import azkaban.imagemgmt.models.ImageType;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ImageTypeDaoImpl implements ImageTypeDao {
+  private static final Logger log = LoggerFactory.getLogger(ImageTypeDaoImpl.class);
+
+  private final DatabaseOperator databaseOperator;
+
+  static String INSERT_IMAGE_TYPE =
+      "insert into image_types ( type, description, active, deployable, created_on, created_by ) "
+          + "values (?, ?, ?, ?, ?, ?)";
+  static String INSERT_IMAGE_OWNERSHIP =
+      "insert into image_ownerships ( type_id, owner, role, created_on, created_by ) "
+          + "values (?, ?, ?, ?, ?)";
+
+  @Inject
+  public ImageTypeDaoImpl(DatabaseOperator databaseOperator) {
+    this.databaseOperator = databaseOperator;
+  }
+
+  @Override
+  public int createImageType(ImageType imageType) {
+    final SQLTransaction<Integer> insertAndGetSpaceId = transOperator -> {
+      String currentTime = DateTime.now().toLocalDateTime().toString();
+      transOperator.update(INSERT_IMAGE_TYPE, imageType.getType(), imageType.getDescription(),
+          true, imageType.getDeployable().getName(), currentTime, imageType.getCreatedBy());
+      int imageTypeId = Long.valueOf(transOperator.getLastInsertId()).intValue();
+      if( imageType.getOwnerships() != null && imageType.getOwnerships().size() > 0) {
+        for(ImageOwnership imageOwnership : imageType.getOwnerships()) {
+          transOperator.update(INSERT_IMAGE_OWNERSHIP, imageTypeId, imageOwnership.getOwner(),
+              imageOwnership.getRole(), currentTime, imageType.getCreatedBy());
```

Comment:
Can we please have enum for all possible Role?

Reply:
Role is added as Enum in the definition file.